src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -185,14 +187,14 @@ public bool TryParseEofHeader(ReadOnlySpan<byte> container, out EofHeader? heade
                 return false;
             }
 
-            int codeSectionsSize = 0;
+            int codeSectionsSize = typeSection.EndOffset;
             SectionHeader[] codeSections = new SectionHeader[numberOfCodeSections];
             for (ushort i = 0; i < numberOfCodeSections; i++)
             {
                 int currentCodeSizeOffset = CODESIZE_OFFSET + i * TWO_BYTE_LENGTH; // offset of i'th code size
                 SectionHeader codeSection = new()
                 {
-                    Start = currentCodeSizeOffset,
+                    Start = codeSectionsSize,
```

Comment:
Can this be `typeSection.EndOffset + codeSectionsSize` ? This way we avoid magic `pos` again.